### DISCUSSION TITLE
An error occurred when running the example case of stoichiometric reactor

### DISCUSSION THREAD

**Discussion by reactingflow** [IDAES/idaes-pse#1372] [Q&A] [answered]
Hi, an error was reported that there was no module named 'egprod_ideal' when i was running the test case of stoichiometric reactor (examples-main/idaes_examples/notebooks/docs/unit_models/reactors/stoichiometric_reactor.ipynb). Can you please tell me how to solve this problem? Thank you.

![image](https://github.com/IDAES/idaes-pse/assets/151173251/3dbb8f00-480f-43da-96b6-465f9327a160)

**Comment by andrewlee94** [comment, marked as answer]
@reactingflow We will need some more information on how you ran the notebook to determine what is happening. I just tested the notebook on my computer and I do not get any errors when running it. Could you send us the list of steps you took to get to this error, including information on the folder you are running from and the environment you are using?

Also, make sure that the files `egprod_ideal.py` and `egprod_reaction.py` are in the same folder as the notebook (if you have copied the notebook somewhere else you will need to copy those files as well.

**Reply by reactingflow** [reply]
Hi, Andrew. @andrewlee94 

Thank you for reply. Your answer is very helpful. By adding these two files into the same folder where the notebook script is located, the problem is solved. 

It seems that these two extra files define the thermodynamic and reaction properties of components involved, respectively, which may be not included in IDAES. So, I would like to know if there is a list of thermodynamic and reaction property modules which have been included in IDAES. It may help us quickly configure the property modules in our practice.

Thank you!

**Reply by andrewlee94** [reply]
There are a number of existing thermophysical property packages in IDAES for general cases which can be found here: https://github.com/IDAES/idaes-pse/tree/main/idaes/models/properties. There is also a Modular Reaction Package tool in there which can build reaction packages for simple (mostly power law type) reactions.

However, properties are often very case specific (both in what properties you need and how you wish to calculate them), and IDAES is primarily built around the concept of letting the user define exactly what they want to do for these. Thus, case-specific property packages, such as those used in the example, are generally encouraged.
